### PR TITLE
Convert continuous and discrete distribution parameters to floatX or int32

### PR DESCRIPTION
**Pull request by lucianopaz**
This should solve issue #3223. While I was working on PR #3293 I also ran into this problem but decided to submit a separate PR for it. I only explicitly casted to `floatX` or `int32` if the distribution's docstring said that the parameter should be a float or an int. The rest (multivariate, mixture and timeseries) I left unchanged.

### CODE REVIEW

**Comment by twiecki** [comment]
Needs a rebase.

**Comment by lucianopaz** [comment]
@twiecki, some tests are failing when `FLOATX='float32'`. Should we mark those as expected failures?

**Comment by lucianopaz** [comment]
By chance I saw #2366, where it says that `int32` should be used if `floatX` is `float64`, and that `int16` should be used if `float32` was set as `floatX`. Maybe we should do the same here.

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -2923,9 +2922,9 @@ def __init__(self, mu=0., sigma=None, nu=None, sd=None,
         if sd is not None:
             sigma = sd
 
-        self.mu = mu = tt.as_tensor_variable(mu)
-        self.sigma = self.sd = sigma = tt.as_tensor_variable(sigma)
-        self.nu = nu = tt.as_tensor_variable(nu)
+        self.mu = mu = tt.as_tensor_variable(floatX(mu))
+        self.sigma = sigma = tt.as_tensor_variable(floatX(sigma))
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]:
```suggestion
        self.sigma = self.sd = sigma = tt.as_tensor_variable(floatX(sigma))
```

File: pymc3/tests/test_distributions.py [hunk 2]
```diff
@@ -1298,11 +1295,11 @@ def setup_class(self):
             Y_obs = Normal('Y_obs', mu=mu, sigma=sigma, observed=Y)
         self.distributions = [alpha, sigma, mu, b, Y_obs]
         self.expected = (
-            r'$\text{alpha} \sim \text{Normal}(\mathit{mu}=0,~\mathit{sigma}=10.0)$',
-            r'$\text{sigma} \sim \text{HalfNormal}(\mathit{sigma}=1.0)$',
+            r'$\text{alpha} \sim \text{Normal}(\mathit{mu}=0.0,~\mathit{sd}=10.0)$',
+            r'$\text{sigma} \sim \text{HalfNormal}(\mathit{sd}=1.0)$',
```
Comment (twiecki) [inline pymc3/tests/test_distributions.py]:
```suggestion
            r'$\text{sigma} \sim \text{HalfNormal}(\mathit{sigma}=1.0)$',
```

File: pymc3/tests/test_distributions.py [hunk 3]
```diff
@@ -1298,11 +1295,11 @@ def setup_class(self):
             Y_obs = Normal('Y_obs', mu=mu, sigma=sigma, observed=Y)
         self.distributions = [alpha, sigma, mu, b, Y_obs]
         self.expected = (
-            r'$\text{alpha} \sim \text{Normal}(\mathit{mu}=0,~\mathit{sigma}=10.0)$',
-            r'$\text{sigma} \sim \text{HalfNormal}(\mathit{sigma}=1.0)$',
+            r'$\text{alpha} \sim \text{Normal}(\mathit{mu}=0.0,~\mathit{sd}=10.0)$',
```
Comment (twiecki) [inline pymc3/tests/test_distributions.py]:
```suggestion
            r'$\text{alpha} \sim \text{Normal}(\mathit{mu}=0.0,~\mathit{sigma}=10.0)$',
```

File: pymc3/tests/test_distributions.py [hunk 4]
```diff
@@ -1298,11 +1295,11 @@ def setup_class(self):
...
+            r'$\text{sigma} \sim \text{HalfNormal}(\mathit{sd}=1.0)$',
             r'$\text{mu} \sim \text{Deterministic}(\text{alpha},~\text{Constant},~\text{beta})$',
-            r'$\text{beta} \sim \text{Normal}(\mathit{mu}=0,~\mathit{sigma}=10.0)$',
-            r'$\text{Y_obs} \sim \text{Normal}(\mathit{mu}=\text{mu},~\mathit{sigma}=f(\text{sigma}))$'
+            r'$\text{beta} \sim \text{Normal}(\mathit{mu}=0.0,~\mathit{sd}=10.0)$',
```
Comment (twiecki) [inline pymc3/tests/test_distributions.py]:
```suggestion
            r'$\text{beta} \sim \text{Normal}(\mathit{mu}=0.0,~\mathit{sigma}=10.0)$',
```

File: pymc3/tests/test_distributions.py [hunk 5]
```diff
@@ -1298,11 +1295,11 @@ def setup_class(self):
...
+            r'$\text{Y_obs} \sim \text{Normal}(\mathit{mu}=f(\text{mu}),~\mathit{sd}=f(\text{sigma}))$'
```
Comment (twiecki) [inline pymc3/tests/test_distributions.py]:
```suggestion
            r'$\text{Y_obs} \sim \text{Normal}(\mathit{mu}=f(\text{mu}),~\mathit{sigma}=f(\text{sigma}))$'
```

**Comment by twiecki** [comment]
```

    def test__repr_latex_(self):
        for distribution, tex in zip(self.distributions, self.expected):
>           assert distribution._repr_latex_() == tex
E           AssertionError: assert '$\\text{Y_ob...ext{sigma}))$' == '$\\text{Y_obs...ext{sigma}))$'
E             - $\text{Y_obs} \sim \text{Normal}(\mathit{mu}=\text{mu},~\mathit{sigma}=f(\text{sigma}))$
E             + $\text{Y_obs} \sim \text{Normal}(\mathit{mu}=f(\text{mu}),~\mathit{sigma}=f(\text{sigma}))$
E             ?                                              ++         +
```

**Comment by twiecki** [comment]
Thanks @lucianopaz!

**Pull request merged**